Objects/typeobject.c

```diff
@@ -3805,11 +3806,15 @@ object_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         Py_DECREF(sorted_methods);
         if (joined == NULL)
             return NULL;
+        method_count = PyObject_Length(abstract_methods);
```

Comment:
I suggest you base `method_count` on `sorted_methods`, not `abstract_methods`, since it's `sorted_methods` that's used to produce `joined`. I realize it doesn't make much difference, but I think it's easier to reason about. You'd need to move this line to before where you `Py_DECREF(sorted_methods)`.

Reply:
Done.